# 🧹 Data Cleaning Project: Big Five Personality Dataset.

This is a project to show a full data cleaning workflow on a large dataset with around 1M rows. the goal is to produce clean and analysis ready dataset.

In [3]:
import pandas as pd

#Load the dataset
file_path = "/Users/mona.moghadam@schibsted.com/Desktop/portfolio, data cleaning/data-final.csv"
data = pd.read_csv(file_path, sep= '\t')
#Quick look at the data
print(data.head())

   EXT1  EXT2  EXT3  EXT4  EXT5  EXT6  EXT7  EXT8  EXT9  EXT10  ...  \
0   4.0   1.0   5.0   2.0   5.0   1.0   5.0   2.0   4.0    1.0  ...   
1   3.0   5.0   3.0   4.0   3.0   3.0   2.0   5.0   1.0    5.0  ...   
2   2.0   3.0   4.0   4.0   3.0   2.0   1.0   3.0   2.0    5.0  ...   
3   2.0   2.0   2.0   3.0   4.0   2.0   2.0   4.0   1.0    4.0  ...   
4   3.0   3.0   3.0   3.0   5.0   3.0   3.0   5.0   3.0    4.0  ...   

              dateload  screenw  screenh  introelapse  testelapse  endelapse  \
0  2016-03-03 02:01:01    768.0   1024.0          9.0       234.0          6   
1  2016-03-03 02:01:20   1360.0    768.0         12.0       179.0         11   
2  2016-03-03 02:01:56   1366.0    768.0          3.0       186.0          7   
3  2016-03-03 02:02:02   1920.0   1200.0        186.0       219.0          7   
4  2016-03-03 02:02:57   1366.0    768.0          8.0       315.0         17   

   IPC  country  lat_appx_lots_of_err  long_appx_lots_of_err  
0    1       GB              

# Initial Data Exploration
-check basic information
-Look for missing values
-Undrestand data types

In [ ]:
#check basic information
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1015341 entries, 0 to 1015340
Columns: 110 entries, EXT1 to long_appx_lots_of_err
dtypes: float64(104), int64(2), object(4)
memory usage: 852.1+ MB
None


In [ ]:
#Look for missing values
print(data.isnull().sum())

EXT1                     1783
EXT2                     1783
EXT3                     1783
EXT4                     1783
EXT5                     1783
                         ... 
endelapse                   0
IPC                         0
country                    77
lat_appx_lots_of_err        0
long_appx_lots_of_err       0
Length: 110, dtype: int64


In [4]:
# undrestad data types
data.sample(5)

EXT1  EXT2  EXT3  EXT4  EXT5  EXT6  EXT7  EXT8  EXT9  EXT10  ...  \
38081    5.0   1.0   3.0   1.0   4.0   2.0   5.0   5.0   1.0    3.0  ...   
662465   4.0   3.0   4.0   3.0   3.0   4.0   3.0   4.0   3.0    3.0  ...   
683996   3.0   5.0   2.0   5.0   1.0   3.0   1.0   3.0   3.0    1.0  ...   
757876   1.0   5.0   1.0   5.0   1.0   4.0   1.0   5.0   2.0    5.0  ...   
908331   5.0   1.0   5.0   5.0   5.0   1.0   5.0   5.0   5.0    3.0  ...   

                   dateload  screenw  screenh  introelapse  testelapse  \
38081   2016-04-03 10:41:52   1440.0    900.0          3.0       161.0   
662465  2018-03-21 05:01:36   1920.0   1080.0          2.0       320.0   
683996  2018-04-02 17:05:34   1536.0    864.0         45.0       136.0   
757876  2018-05-23 17:20:47    412.0    732.0          4.0       427.0   
908331  2018-09-11 03:56:57    768.0   1024.0          3.0       384.0   

        endelapse  IPC  country  lat_appx_lots_of_err  long_appx_lots_of_err  
38081          12    1       US                40.064               -80.7209  
662465         18    1       CY               35.1667                33.3667  
683996         11    1       US               41.9847               -88.0798  
757876         12    2       US               41.7804               -87.6027  
908331         14   43       IE               53.3478                -6.2597  

[5 rows x 110 columns]

# Data Cleaning:

## 1. Handling Missing values

In [5]:
# fill missing countries with Unknown
data['country'].fillna('Unknown', inplace=True)

/var/folders/6x/xl2p1y_d5hjfmd70sn4y94lr0000gn/T/ipykernel_46736/718640259.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data['country'].fillna('Unknown', inplace=True)


## 2. Dropping Incomplete Survey Responses

In [6]:
#remove rows with missing answers
survey_cols = [col for col in data.columns if col.startswith(('EXT', 'EST', 'AGR', 'CSN', 'OPN'))]
data.dropna(subset= survey_cols, inplace = True)
print('Remaining rows: ', len(data))

Remaining rows:  1013558


## 3. Converting Data types

In [ ]:
#Convert 'dateload' to datetime
data['dateload'] = pd.to_datetime(data['dateload'], errors='coerce')
data.drop_duplicates(inplace= True)

In [ ]:
#Convert latitude and longitude to numeric values
data['lat_appx_lots_of_err'] = pd.to_numeric(data['lat_appx_lots_of_err'], errors = 'coerce')
data['long_appx_lots_of_err'] = pd.to_numeric(data['long_appx_lots_of_err'], errors = 'coerce')

print(data[['lat_appx_lots_of_err', 'long_appx_lots_of_err']].dtypes)

lat_appx_lots_of_err     float64
long_appx_lots_of_err    float64
dtype: object


# Final Data Validation:
- Check missing values
- Confirm data types
- Sample final rows

In [ ]:
#check for missing values after cleaning
print(data.isnull().sum())

EXT1                         0
EXT2                         0
EXT3                         0
EXT4                         0
EXT5                         0
                         ...  
endelapse                    0
IPC                          0
country                      0
lat_appx_lots_of_err     13721
long_appx_lots_of_err    13721
Length: 110, dtype: int64


In [ ]:
#Confirm data types after cleaning
print(data.dtypes)

EXT1                     float64
EXT2                     float64
EXT3                     float64
EXT4                     float64
EXT5                     float64
                          ...   
endelapse                  int64
IPC                        int64
country                   object
lat_appx_lots_of_err     float64
long_appx_lots_of_err    float64
Length: 110, dtype: object


In [ ]:
#sample final rows of clean data
print(data.sample(5))


        EXT1  EXT2  EXT3  EXT4  EXT5  EXT6  EXT7  EXT8  EXT9  EXT10  ...  \
259050   4.0   4.0   4.0   2.0   3.0   2.0   4.0   3.0   4.0    2.0  ...   
973388   4.0   2.0   4.0   2.0   4.0   2.0   4.0   4.0   4.0    2.0  ...   
496694   3.0   3.0   4.0   3.0   4.0   2.0   2.0   3.0   4.0    2.0  ...   
278418   1.0   5.0   1.0   5.0   1.0   5.0   1.0   5.0   2.0    5.0  ...   
448317   2.0   3.0   3.0   3.0   3.0   4.0   2.0   4.0   2.0    3.0  ...   

                   dateload  screenw  screenh  introelapse  testelapse  \
259050  2016-11-06 16:07:11   1920.0   1080.0          3.0       399.0   
973388  2018-10-17 07:19:26   1280.0    720.0         23.0       295.0   
496694  2017-09-24 13:09:14   1280.0    800.0         88.0       341.0   
278418  2016-11-30 03:52:34   1440.0    900.0          4.0       204.0   
448317  2017-07-10 11:08:40    768.0   1024.0          2.0       119.0   

        endelapse  IPC  country  lat_appx_lots_of_err  long_appx_lots_of_err  
259050        213  

# Save the Clean Dataset

In [ ]:
data.to_csv("/Users/mona.moghadam@schibsted.com/Desktop/portfolio, data cleaning/data-final_cleaned.csv", index=False)
print("✅ Cleaned data saved successfully!")

# 📈 Summary of Cleaning

- Loaded raw tab-separated survey dataset (1M+ entries)
- Filled missing values for 'country'
- Dropped 1,783 incomplete survey responses
- Converted dates and geographical coordinates to correct formats
- Verified clean, complete dataset
- Final dataset ready for analysis or modeling

Tools used: **Python, Pandas, Jupyter Notebook**